src/components/header/header.component.tsx

```diff
@@ -0,0 +1,145 @@
+import React, { useCallback, useEffect, useState } from 'react';
+import { HeaderTypes } from './header.types';
+import { addTodo, USER_ID } from '../../api/todos';
+import { Todo } from '../../types/Todo';
+import { text } from '../../constants/text';
+
+export const HeaderComponent: React.FC<HeaderTypes> = ({
+  setCustomError,
+  setTodos,
+  customError,
+  handleLoaderId,
+  titleField,
+}) => {
+  const [disabled, setIsDisabled] = useState<boolean>(false);
+  const [newTodo, setNewTodo] = useState<Todo>({
+    id: 0,
+    userId: USER_ID,
+    title: '',
+    completed: false,
+  });
+
+  useEffect(() => {
+    if (titleField.current) {
+      titleField.current.focus();
+    }
+  }, [titleField]);
+
+  const titleHandleChange = useCallback(
+    (event: React.ChangeEvent<HTMLInputElement>) => {
+      setNewTodo(prevState => ({
+        ...prevState,
+        title: event.target.value,
+      }));
+    },
+    [],
+  );
+
+  const reset = useCallback(() => {
+    setNewTodo(prevState => ({ ...prevState, title: '' }));
+  }, []);
+
+  const onSubmit = useCallback(
```

Comment:
basically you dont need use callback here, check in witch cases you need to use it